### PR TITLE
Add superform

### PR DESCRIPTION
**Pull request by danieladugyan**

### CODE REVIEW

File: src/routes/positions/[id]/DeleteMandateForm.svelte [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    id: mandateId,
+  });
+</script>
+
+<form action="?/deleteMandate" method="POST" use:enhance>
+  <input type="hidden" name="mandateId" value={mandateId} />
+  <button
+    type="submit"
+    class="btn btn-error btn-sm pointer-events-auto"
+    on:click|stopPropagation={() => {}}
```
Comment (danieladugyan) [inline src/routes/positions/[id]/DeleteMandateForm.svelte:18]: Why? (preferreably in a comment)

File: src/routes/news/likes.ts [hunk 2]
```diff
@@ -1,51 +1,43 @@
...
+                studentId: session?.user?.student_id,
+              },
+            },
+          },
+          select: {
+            id: true,
+          },
+        });
+        return message(form, {
+          message: `${shouldLike ? "Gillat" : "Slutat gilla"} artikel`,
```
Comment (danieladugyan) [inline src/routes/news/likes.ts:37]: Just include `artikel` in both of the strings, makes extracting them into translations so much easier.

File: src/routes/news/create/+page.server.ts [hunk 3]
```diff
@@ -107,6 +95,9 @@ export const actions = {
                           },
                         }
                       : undefined,
+                    customAuthor: author.customId
+                      ? { connect: { id: author.customId } }
+                      : undefined,
                   },
             },
             // imageUrl: (image as string | undefined) || null,
```
Comment (danieladugyan) [inline src/routes/news/create/+page.server.ts:103]: Remove

File: src/routes/news/[slug]/edit/+page.server.ts [hunk 4]
```diff
@@ -52,111 +56,102 @@ export const load: PageServerLoad = async ({ parent, params }) => {
...
-                        studentId: session!.user.student_id,
+        });
+        try {
+          await prisma.article.update({
+            where: {
+              slug: slug,
+            },
+            data: {
+              header: header,
+              body: body,
```
Comment (danieladugyan) [inline src/routes/news/[slug]/edit/+page.server.ts:95]: abbreviate?

File: src/routes/members/[studentId]/profile-picture/+page.server.ts [hunk 5]
```diff
@@ -28,139 +29,141 @@ export const load: PageServerLoad = async ({ params, parent }) => {
...
+            .resize(400, 400, {
+              fit: "cover",
+              withoutEnlargement: true,
+            })
+            // save as webp
+            .webp()
+            .toBuffer();
+          const putUrl = await fileHandler.getPresignedPutUrl(
+            session?.user,
+            "dev-members",
```
Comment (danieladugyan) [inline src/routes/members/[studentId]/profile-picture/+page.server.ts:114]: Make this an ENV variable

File: src/routes/documents/upload/+page.server.ts [hunk 6]
```diff
@@ -1,62 +1,66 @@
...
-      if ("body" in e && "message" in e.body) {
-        return failWithData(e.body.message as string);
-      }
-      return failWithData(String(e));
-    }
+    const form = await superValidate(formData, uploadSchema);
+    if (!form.valid) return fail(400, { form });
+    const session = await locals.getSession();
+    return withAccess(
+      apiNames.FILES.BUCKET("dev-documents").CREATE,
```
Comment (danieladugyan) [inline src/routes/documents/upload/+page.server.ts:34]: Replace with env

File: src/routes/documents/upload/+page.server.ts [hunk 7]
```diff
@@ -1,62 +1,66 @@
...
+      session?.user,
+      async () => {
+        const { meeting, name, year } = form.data;
+        const file = formData.get("file");
+        if (!file || !(file instanceof File) || file.size <= 0) {
+          return setError(form, "file", "Felaktig fil");
+        }
+        const formattedName =
+          String(name)
+            .replace(/\s/g, "_")
```
Comment (danieladugyan) [inline src/routes/documents/upload/+page.server.ts:44]: Replace with a function call to make it easier to understand what's going on

File: src/routes/committees/[shortName]/+page.server.ts [hunk 8]
```diff
@@ -78,72 +88,71 @@ export const load: PageServerLoad = async ({ params }) => {
...
+      session?.user,
+      async () => {
+        const image = formData.get("image");
+        if (
+          !image ||
+          !(image instanceof File) ||
+          image.size <= 0 ||
+          image.type !== "image/svg+xml"
+        ) {
+          return setError(form, "image", "Du måste ladda upp en korrekt bild");
```
Comment (danieladugyan) [inline src/routes/committees/[shortName]/+page.server.ts:118]: Could this error perhaps be more helpful? Is the svg requirement reflected in the frontend file upload. If not it should probably be reflected in this error.

File: src/routes/committees/[shortName]/+page.server.ts [hunk 9]
```diff
@@ -78,72 +88,71 @@ export const load: PageServerLoad = async ({ params }) => {
...
-        return fail(500, {
-          error: e.message ?? "Unknown error",
-          data: Object.fromEntries(formData),
+        let newImageUploaded = false;
+        const path = `committees/${params.shortName}.svg`;
+        if (image) {
+          try {
+            const putUrl = await fileHandler.getPresignedPutUrl(
+              session?.user,
+              "dev-material",
```
Comment (danieladugyan) [inline src/routes/committees/[shortName]/+page.server.ts:126]: Replace with env

**Review by Macludde — APPROVED** [review]

**Pull request merged**